contributors/design-proposals/local-pv-provisioner.md

```diff
@@ -0,0 +1,92 @@
+# Local PV Distributed Static Provisioner
+Authors: @verult, @msau42
+        
+## Design Goals
+1. Limit all PV object interactions to a single master pod in order to minimize node access to the Kubernetes system.
+1. Assume disk deletion is asynchronous, as required by block volumes.
+
+## Related Works
+* Original provisioner design, [here](https://github.com/kubernetes-incubator/external-storage/tree/master/local-volume/provisioner#design).
+* Disk cleaner (deleter) implementation for block volumes, [here](https://github.com/kubernetes-incubator/external-storage/pull/312).
+
+## Design 1: Master-Heavy
+
+### High Level Overview
+The static provisioner will be split into a single master (ideally placed on the master node) and many minions (one per node). The master contains the bulk of provisioner logic, including syncing PV objects with the API server and triggering discovery and deletion. Minions are left to perform the physical discovery and deletion of the local volumes on its node. gRPC is used to communicate between master and minions, with the master as the client and each minion as a separate server.
+
+The master keeps track of `MinionInfo`, a collection containing one entry for each minion keeping track of relevant information. The master also runs a group of goroutines, one for each minion, which controls and interacts with the minions.
```

Comment:
Minion terminology is problematic for historical reasons because we have been trying to be consistent.  I assume you're referring to an agent that runs on the node that is tracking volume state?  Why not call it "node volume detector" and be consistent with "node problem detector"?

I don't like using minion because it's too generic and it overlaps too heavily with our historical work around node, and it's not clear that this is a very specific agent.

Reply:
ACK, will update in a later iteration.